pkg/kubernetes/deployment/extnv1beta1/v1alpha1/kubernetes.go

```diff
@@ -0,0 +1,187 @@
+func (k *kubeclient) withDefaults() {
+	if k.getClientset == nil {
+		k.getClientset = func() (clients *kubernetes.Clientset, err error) {
+			config, err := client.GetConfig(client.New())
+			if err != nil {
+				return nil, err
+			}
+			return kubernetes.NewForConfig(config)
+		}
+	}
+
+	if k.get == nil {
+		k.get = func(cli *kubernetes.Clientset, name,
+			namespace string, opts *metav1.GetOptions) (
+			d *extnv1beta1.Deployment, err error) {
+			d, err = cli.ExtensionsV1beta1().
+				Deployments(namespace).
+				Get(name, *opts)
+			return
+		}
+	}
+
+	if k.rolloutStatus == nil {
+		k.rolloutStatus = func(d *extnv1beta1.Deployment) (
+			*rolloutOutput, error) {
+			status, err := New(WithAPIObject(d)).
+				RolloutStatus()
+			if err != nil {
+				return nil, err
+			}
+			return NewRollout(
+				withOutputObject(status)).
+				AsRolloutOutput()
+		}
+	}
+
+	if k.rolloutStatusf == nil {
+		k.rolloutStatusf = func(d *extnv1beta1.Deployment) (
+			[]byte, error) {
+			status, err := New(WithAPIObject(d)).
+				RolloutStatus()
+			if err != nil {
+				return nil, err
+			}
+			return NewRollout(
+				withOutputObject(status)).
+				Raw()
+		}
+	}
+
+}
+
+// WithClientset sets the kubernetes client against the kubeclient instance
+func WithClientset(c *kubernetes.Clientset) kubeclientBuildOption {
+	return func(k *kubeclient) {
+		k.clientset = c
+	}
+}
+
+// WithNamespace set namespace in kubeclient object
+func WithNamespace(namespace string) kubeclientBuildOption {
+	return func(k *kubeclient) {
+		k.namespace = namespace
+	}
+}
+
+// KubeClient returns a new instance of kubeclient meant for deployment.
+// caller can configure it with different kubeclientBuildOption
+func KubeClient(opts ...kubeclientBuildOption) *kubeclient {
+	k := &kubeclient{}
+	for _, o := range opts {
+		o(k)
+	}
+	k.withDefaults()
+	return k
+}
+
+// getClientOrCached returns either a new
+// instance of kubernetes client or its cached copy
+func (k *kubeclient) getClientOrCached() (*kubernetes.Clientset, error) {
+	if k.clientset != nil {
+		return k.clientset, nil
+	}
+	c, err := k.getClientset()
+	if err != nil {
+		return nil, err
+	}
+	k.clientset = c
+	return k.clientset, nil
+}
+
+// Get returns deployment object for given name
+func (k *kubeclient) Get(name string) (*extnv1beta1.Deployment, error) {
+	cli, err := k.getClientOrCached()
+	if err != nil {
+		return nil, err
+	}
+	return k.get(cli, name, k.namespace, &metav1.GetOptions{})
+}
+
+// RolloutStatus returns deployment's rollout status for given name
+func (k *kubeclient) RolloutStatus(name string) (
+	*rolloutOutput, error) {
+	cli, err := k.getClientOrCached()
+	if err != nil {
+		return nil, err
+	}
+	d, err := k.get(cli, name, k.namespace, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return k.rolloutStatus(d)
+}
+
+// RolloutStatusf returns deployment's rollout status for given name
+// in raw bytes
+func (k *kubeclient) RolloutStatusf(name string) (
```

Comment:
let us have the return args in the same line as start of function definition

Reply:
done